examples/backup_vm.py

```diff
@@ -338,47 +354,32 @@ def start_backup(connection, args):
 
     if backup.to_checkpoint_id is not None:
         progress(
-            "Created checkpoint %r (to use in --from-checkpoint-uuid "
-            "for the next incremental backup)" % backup.to_checkpoint_id)
+            "Created checkpoint %r" % backup.to_checkpoint_id)
 
     return backup
 
 
-def stop_backup(connection, backup_uuid, args):
-    verify_vm_exists(connection, args.vm_uuid)
-    verify_backup_exists(connection, args.vm_uuid, backup_uuid)
-
-    backup_service = get_backup_service(connection, args.vm_uuid, backup_uuid)
+def stop_backup(connection, backup, args):
+    backup_service = get_backup_service(connection, backup.vm.id, backup.id)
 
     backup_service.finalize()
 
     # "get_backup()" invocation will raise if the backup failed.
     # So we just need to wait until the backup phase is SUCCEEDED.
-    backup = get_backup(connection, backup_service, backup_uuid)
     while backup.phase != types.BackupPhase.SUCCEEDED:
         time.sleep(1)
-        backup = get_backup(connection, backup_service, backup_uuid)
+        backup = get_backup(connection, backup_service, backup.id)
 
 
-def download_backup(connection, backup_uuid, args, incremental=False):
-    verify_vm_exists(connection, args.vm_uuid)
-    verify_backup_exists(connection, args.vm_uuid, backup_uuid)
-
+def download_backup(connection, backup, args, incremental=False):
     if not args.download_backup:
         progress("Skipping download")
         return
 
-    backup_service = get_backup_service(connection, args.vm_uuid, backup_uuid)
-
-    # "get_backup()" invocation will raise if the backup failed.
-    # So we just need to verify that the backup phase is READY.
-    backup = get_backup(connection, backup_service, backup_uuid)
-    if backup.phase != types.BackupPhase.READY:
```

Comment:
What if `download_backup` will be called from `cmd_full` or `cmd_incremental`, isn't it a problem that there is not checking backup phase?
After the check for `BackupPhase.READY` was moved to `cmd_download`?

Reply:
cmd_full and cmd_incremental wait until backup is ready before they download the backup.